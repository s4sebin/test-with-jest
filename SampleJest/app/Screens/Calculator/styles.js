import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',

    backgroundColor: 'white',
  },
  buttonContainer: {
    padding: 8,
    paddingHorizontal : 12,
    margin: 10,
    borderRadius: 5,
    alignSelf: 'center',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'grey',
  },
  title: {
    marginBottom: 32,

    color: 'black',
    fontSize: 32,
    fontWeight: 'bold',
  },
  iconsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',

    width: '100%',
    maxWidth: 400,

    marginVertical: 16,
  },
  safeareaview: {
    flex: 1,
    backgroundColor: 'white',
  },
});
