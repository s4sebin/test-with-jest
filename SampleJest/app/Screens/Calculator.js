import React, {useEffect, useState} from 'react';
import {Alert, Text, TouchableOpacity, View} from 'react-native';
import styles from './styles.js';
import {Input} from '../components/Input.js';

const OperationsConst = {
  SUM: 'SUM',
  SUBTRACTION: 'SUBTRACTION',
  MULTIPLICATION: 'MULTIPLICATION',
  DIVISION: 'DIVISION',
};

export default function Calculator() {
  function clearResult() {
    setOperation(null);
    setResult(0);
  }

  function handleSum() {
    setOperation(OperationsConst.SUM);
    setResult(Number(firstValue) + Number(secondValue));
  }

  function handleSubtraction() {
    setOperation(OperationsConst.SUBTRACTION);
    setResult(Number(firstValue) - Number(secondValue));
  }

  function handleMultiplication() {
    setOperation(OperationsConst.MULTIPLICATION);
    setResult(Number(firstValue) * Number(secondValue));
  }

  function handleDivision() {
    setOperation(OperationsConst.DIVISION);

    if (secondValue === 0) {
      Alert.alert('Aviso', 'Não pode fazer divisão por 0 (zero)!!!');
    } else {
      setResult(Number(firstValue) / Number(secondValue));
    }
  }

  let [firstValue, setFirstValue] = useState(0);
  let [secondValue, setSecondValue] = useState(0);
  let [result, setResult] = useState(0);
  let [operation, setOperation] = useState(null);

  useEffect(() => {
    clearResult();
  }, [firstValue, secondValue]);

  return (
    <View style={styles.container}>
      <Text style={styles.title}>CALCULATOR</Text>

      <Input
        editable={true}
        label="PRIMARY VALUE"
        value={firstValue}
        onChangeText={text => setFirstValue(text)}
      />

      <Input
        editable={true}
        label="SECONDARY VALUE"
        value={secondValue}
        onChangeText={text => setSecondValue(text)}
      />

      <View style={styles.iconsContainer}>
        <TouchableOpacity style={styles.buttonContainer} onPress={handleSum} testID="sumButton">
          <Text>Add</Text>
        </TouchableOpacity>

        <TouchableOpacity
          style={styles.buttonContainer}
          onPress={handleSubtraction}>
          <Text>Sub</Text>
        </TouchableOpacity>

        <TouchableOpacity
          style={styles.buttonContainer}
          onPress={handleMultiplication}>
          <Text>Mul</Text>
        </TouchableOpacity>

        <TouchableOpacity
          style={styles.buttonContainer}
          onPress={handleDivision}>
          <Text>Div</Text>
        </TouchableOpacity>
      </View>

      <Input centralize={true} editable={false} label="RESULT" value={result} />
    </View>
  );
}
