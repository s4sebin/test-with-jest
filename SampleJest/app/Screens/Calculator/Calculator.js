import React, {useEffect, useState} from 'react';
import {Alert, Text, TouchableOpacity, View} from 'react-native';
import styles from './styles.js';
import { Input } from '../../components/Input.js';

const OperationsConst = {
  SUM: 'SUM',
  SUBTRACTION: 'SUBTRACTION',
  MULTIPLICATION: 'MULTIPLICATION',
  DIVISION: 'DIVISION',
};

export default function Calculator() {
  function clearResult() {
    setResult(0);
  }

  function handleSum() {
    setResult(Number(firstValue) + Number(secondValue));
  }

  function handleSubtraction() {
    setResult(Number(firstValue) - Number(secondValue));
  }

  function handleMultiplication() {
    setResult(Number(firstValue) * Number(secondValue));
  }

  function handleDivision() {
    if (secondValue === 0) {
      Alert.alert('Errorr! Number cannot divided by 0');
    } else {
      setResult(Number(firstValue) / Number(secondValue));
    }
  }

  let [firstValue, setFirstValue] = useState(0);
  let [secondValue, setSecondValue] = useState(0);
  let [result, setResult] = useState(0);

  useEffect(() => {
    clearResult();
  }, [firstValue, secondValue]);

  return (
    <View style={styles.container}>
      <Text style={styles.title}>CALCULATOR</Text>

      <Input
        testID="testPrimaryValue"
        editable={true}
        label="PRIMARY VALUE"
        value={firstValue}
        onChangeText={text => setFirstValue(text)}
      />

      <Input
        testID="testSecondaryValue"
        editable={true}
        label="SECONDARY VALUE"
        value={secondValue}
        onChangeText={text => setSecondValue(text)}
      />

      <View style={styles.iconsContainer}>
        <TouchableOpacity
          style={styles.buttonContainer}
          onPress={handleSum}
          testID="sumButton">
          <Text>Add</Text>
        </TouchableOpacity>

        <TouchableOpacity
          style={styles.buttonContainer}
          onPress={handleSubtraction}
          testID="subButton">
          <Text>Sub</Text>
        </TouchableOpacity>

        <TouchableOpacity
          style={styles.buttonContainer}
          onPress={handleMultiplication}
          testID="mulButton">
          <Text>Mul</Text>
        </TouchableOpacity>

        <TouchableOpacity
          style={styles.buttonContainer}
          testID="divButton"
          onPress={handleDivision}>
          <Text>Div</Text>
        </TouchableOpacity>
      </View>

      <Input
        centralize={true}
        testID="resultValue"
        editable={false}
        label="RESULT"
        value={result}
      />
    </View>
  );
}
