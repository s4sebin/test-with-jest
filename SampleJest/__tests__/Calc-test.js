/**
 * @format
 */

import 'react-native';
import React from 'react';
import Calculator from '../app/Screens/Calculator';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
