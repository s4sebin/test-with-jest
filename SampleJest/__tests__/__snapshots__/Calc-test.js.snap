// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 32,
        "fontWeight": "bold",
        "marginBottom": 32,
      }
    }
  >
    CALCULATOR
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 16,
          "marginBottom": 8,
        }
      }
    >
      PRIMARY VALUE
    </Text>
    <TextInput
      editable={true}
      keyboardType="numeric"
      onChangeText={[Function]}
      style={
        Array [
          undefined,
          Object {
            "borderColor": "black",
            "borderRadius": 8,
            "borderWidth": 1,
            "color": "black",
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 18,
            "padding": 8,
            "width": 160,
          },
          undefined,
          false,
        ]
      }
      value="0"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 16,
          "marginBottom": 8,
        }
      }
    >
      SECONDARY VALUE
    </Text>
    <TextInput
      editable={true}
      keyboardType="numeric"
      onChangeText={[Function]}
      style={
        Array [
          undefined,
          Object {
            "borderColor": "black",
            "borderRadius": 8,
            "borderWidth": 1,
            "color": "black",
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 18,
            "padding": 8,
            "width": 160,
          },
          undefined,
          false,
        ]
      }
      value="0"
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "marginVertical": 16,
        "maxWidth": 400,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "grey",
          "borderRadius": 5,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
          "padding": 8,
          "paddingHorizontal": 12,
        }
      }
    >
      <Text>
        Add
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "grey",
          "borderRadius": 5,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
          "padding": 8,
          "paddingHorizontal": 12,
        }
      }
    >
      <Text>
        Sub
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "grey",
          "borderRadius": 5,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
          "padding": 8,
          "paddingHorizontal": 12,
        }
      }
    >
      <Text>
        Mul
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "grey",
          "borderRadius": 5,
          "justifyContent": "center",
          "margin": 10,
          "opacity": 1,
          "padding": 8,
          "paddingHorizontal": 12,
        }
      }
    >
      <Text>
        Div
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 16,
          "marginBottom": 8,
        }
      }
    >
      RESULT
    </Text>
    <TextInput
      editable={false}
      keyboardType="numeric"
      style={
        Array [
          undefined,
          Object {
            "borderColor": "black",
            "borderRadius": 8,
            "borderWidth": 1,
            "color": "black",
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 18,
            "padding": 8,
            "width": 160,
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "lightgreen",
            "color": "black",
          },
        ]
      }
      value="0"
    />
  </View>
</View>
`;
